bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/gemini/ghost/GhostMixin.java

```diff
@@ -0,0 +1,10 @@
+package edu.gemini.spModel.gemini.ghost;
+
+import edu.gemini.pot.sp.SPComponentType;
+
+// We need to do this because of reflection. This is a static property of instruments, but since Ghost is
+// coded in Scala, values in the companion object are called as methods, which is incompatible with reflection
+// on fields.
+public interface GhostMixin {
+    SPComponentType SP_TYPE = SPComponentType.INSTRUMENT_GHOST;
```

Comment:
This can be read as a static field in `Ghost` from Java classes, but not from Scala, which looks at the companion object when you access `Ghost`. Even changing it to:

`object Ghost extends GhostMixin`

doesn't allow us to access `SP_TYPE` through `Ghost` in Scala, so we need to access `GhostMixin.SP_TYPE` directly, as seen below in `GhostNI.scala`.

Reply:
And the other way around? Maybe put it on `object Ghost`? and call it from Java? or is that the reflection bit?

Reply:
ok, nevermind, it isn't worth the effort. could you make sure this would compile on scala 2.12? We'll have to use these classes from the seqexec

Reply:
@cquiroz or @swalker2m, is there any way you know of to try compiling these files with Scala 2.12? I know OCS2 doesn't compile with it, but I don't know any easy way to try compiling only certain files, especially given how many dependencies `Ghost.scala` has.